src/drivers/ms5525_airspeed/MS5525.cpp

```diff
@@ -0,0 +1,319 @@
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "MS5525.hpp"
+
+int
+MS5525::measure()
+{
+	int ret = PX4_ERROR;
+
+	if (_inited) {
+		// send the command to begin a conversion.
+		uint8_t cmd = _current_cmd;
+		ret = transfer(&cmd, 1, nullptr, 0);
+
+		if (ret != PX4_OK) {
+			perf_count(_comms_errors);
+		}
+
+	} else {
+		_inited = init_ms5525();
+		ret = PX4_OK;
+	}
+
+	return ret;
+}
+
+bool
+MS5525::init_ms5525()
+{
+	// Step 1 - reset
+	uint8_t cmd = CMD_RESET;
+	int ret = transfer(&cmd, 1, nullptr, 0);
+
+	if (ret != PX4_OK) {
+		perf_count(_comms_errors);
+		return false;
+	}
+
+	usleep(3000);
+
+	// Step 2 - read calibration coefficients from prom
+
+	// prom layout
+	// 0 factory data and the setup
+	// 1-6 calibration coefficients
+	// 7 serial code and CRC
+	uint16_t prom[8];
+
+	for (int i = 0; i < 8; i++) {
+		cmd = CMD_PROM_START + i * 2;
+
+		// request PROM value
+		ret = transfer(&cmd, 1, nullptr, 0);
+
+		if (ret != PX4_OK) {
+			perf_count(_comms_errors);
+			return false;
+		}
+
+		// read 2 byte value
+		uint8_t val[2];
+		ret = transfer(nullptr, 0, &val[0], 2);
+
+		if (ret == PX4_OK) {
+			prom[i] = (val[0] << 8) | val[1];
+			// PX4_INFO("request PROM value cmd: %X ret: %04X", cmd, prom[i]);
+
+		} else {
+			perf_count(_comms_errors);
+			return false;
+		}
+	}
+
+	// Step 3 - check CRC
+	uint8_t crc = prom_crc4(prom);
+	uint8_t onboard_crc = prom[7] & 0xF;
+
+	if (crc == onboard_crc && crc != 0) {
```

Comment:
This was basically a hack from initial development when all registers were coming back 0. I can remove it.